kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/engine/DefaultCommitCoordinatorClientHandler.java

```diff
@@ -0,0 +1,164 @@
+            AbstractMetadata currentMetadata,
+            AbstractProtocol currentProtocol) {
+        return commitCoordinatorClient.registerTable(
+                new Path(logPath),
+                currentVersion,
+                StorageKernelAPIAdapter.toStorageAbstractMetadata(currentMetadata),
+                StorageKernelAPIAdapter.toStorageAbstractProtocol(currentProtocol));
+    }
+
+    @Override
+    public CommitResponse commit(
+            String logPath,
+            Map<String, String> tableConf,
+            long commitVersion,
+            CloseableIterator<Row> actions,
+            UpdatedActions updatedActions) {
+        Path path = new Path(logPath);
+        LogStore logStore = LogStoreProvider.getLogStore(hadoopConf, path.toUri().getScheme());
+        return StorageKernelAPIAdapter.toKernelAPICommitResponse(commitCoordinatorClient.commit(
+                logStore,
+                hadoopConf,
+                path,
+                tableConf,
+                commitVersion,
+                new Iterator<String>() {
+                    @Override
+                    public boolean hasNext() {
+                        return actions.hasNext();
+                    }
+
+                    @Override
+                    public String next() {
+                        return JsonUtils.rowToJson(actions.next());
+                    }
+                },
+                StorageKernelAPIAdapter.toStorageUpdatedActions(updatedActions)));
+    }
+
+    @Override
+    public GetCommitsResponse getCommits(
+            String logPath,
+            Map<String, String> tableConf,
+            Long startVersion,
+            Long endVersion) {
+        return StorageKernelAPIAdapter.toKernelAPIGetCommitsResponse(
+                commitCoordinatorClient.getCommits(
+                        new Path(logPath),
+                        tableConf,
+                        startVersion,
+                        endVersion));
+    }
+
+    @Override
+    public void backfillToVersion(
+            String logPath,
+            Map<String, String> tableConf,
+            long version,
+            Long lastKnownBackfilledVersion) {
+        Path path = new Path(logPath);
+        LogStore logStore = LogStoreProvider.getLogStore(hadoopConf, path.toUri().getScheme());
+        commitCoordinatorClient.backfillToVersion(
+                logStore,
+                hadoopConf,
+                path,
+                tableConf,
+                version,
+                lastKnownBackfilledVersion);
+    }
+
+    @Override
+    public boolean semanticEquals(CommitCoordinatorClientHandler other) {
+        return other instanceof DefaultCommitCoordinatorClientHandler &&
+                commitCoordinatorClient.semanticEquals(
+                ((DefaultCommitCoordinatorClientHandler) other).getCommitCoordinatorClient());
+    }
+
+    public CommitCoordinatorClient getCommitCoordinatorClient() {
+        return commitCoordinatorClient;
+    }
```

Comment:
This looks like new API not present in the interfaces. Why do we need to expose the delta-storage client? Add a doc.

Reply:
Thanks for pointing it out. It should be private.